From 8c295296dc185a5da2a3dc6571d9ca0bb3d46a0b Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 13:18:23 -0400
Subject: [ethernet] r8169: Call dev_kfree_skby_any instead of dev_kfree_skb

Message-id: <1410182312-19560-5-git-send-email-ivecera@redhat.com>
Patchwork-id: 91975
O-Subject: [RHEL7.1 PATCH 04/13] r8169: Call dev_kfree_skby_any instead of dev_kfree_skb.
Bugzilla: 1139247
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: #1139247

Upstream commit(s):
commit 989c9ba104d9ce53c1ca918262f3fdfb33aca12a
Author: Eric W. Biederman <ebiederm@xmission.com>
Date:   Tue Mar 11 14:16:14 2014 -0700

    r8169: Call dev_kfree_skby_any instead of dev_kfree_skb.

    Replace dev_kfree_skb with dev_kfree_skb_any in functions that can
    be called in hard irq and other contexts.

    Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 3ff7bc3..90c14d1 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -5834,7 +5834,7 @@ static void rtl8169_tx_clear_range(struct rtl8169_private *tp, u32 start,
           tp->TxDescArray + entry);
    if (skb) {
     tp->dev->stats.tx_dropped++;
-    dev_kfree_skb(skb);
+    dev_kfree_skb_any(skb);
     tx_skb->skb = NULL;
    }
   }
@@ -6059,7 +6059,7 @@ static netdev_tx_t rtl8169_start_xmit(struct sk_buff *skb,
 err_dma_1:
  rtl8169_unmap_tx_skb(d, tp->tx_skb + entry, txd);
 err_dma_0:
- dev_kfree_skb(skb);
+ dev_kfree_skb_any(skb);
 err_update_stats:
  dev->stats.tx_dropped++;
  return NETDEV_TX_OK;
@@ -6142,7 +6142,7 @@ static void rtl_tx(struct net_device *dev, struct rtl8169_private *tp)
    tp->tx_stats.packets++;
    tp->tx_stats.bytes += tx_skb->skb->len;
    u64_stats_update_end(&tp->tx_stats.syncp);
-   dev_kfree_skb(tx_skb->skb);
+   dev_kfree_skb_any(tx_skb->skb);
    tx_skb->skb = NULL;
   }
   dirty_tx++;
-- 
1.7.1